### PR TITLE
feat: Allow aborting uploads using AbortSignal

### PR DESCRIPTION
**Pull request by lucaas**
Resolves FT-4fe346cb-3f20-438d-9b07-d1ee95f19e33

- [x] I have added automatic tests where applicable
- [x] The PR title is suitable as a release note
- [x] The PR contains a description of what has been changed
- [x] The description contains manual test instructions

I branched this off the latest tag as I had some issues testing with the new event hub client implementation.

## Changes

* Deprecate `xhr` option as that is will not be compatible with multi-part uploads
* Introduce a signal option for aborting uploads (similar to other methods)

## Test

1. Check that you agree with API
2. Verify that tests run.

### CODE REVIEW

File: source/session.ts [hunk 1]
```diff
@@ -962,6 +970,10 @@ export class Session {
     let url: string;
     let headers: Record<string, string> = {};
 
+    options.signal?.addEventListener("abort", () => {
+      xhr.abort();
+    });
```
Comment (jimmycallin) [inline source/session.ts:975]: should we remove the event listener as soon as the abort is triggered?

File: source/session.ts [hunk 2]
```diff
@@ -949,6 +951,12 @@ export class Session {
     const defaultProgress = (progress: number) => progress;
     const defaultAbort = () => {};
 
+    if (options.xhr) {
+      logger.warn(
+        "[session.createComponent] options.xhr is deprecated, use options.signal for aborting uploads."
```
Comment (jimmycallin) [inline source/session.ts:956]: please add an issue that mentions this for next major

**Review by jimmycallin — APPROVED** [review]
code and tests looks good, haven't tried it but you can merge it imo once you have considered my question

File: source/session.ts (hunk 1 above)
Comment (lucaas) [reply source/session.ts]: [144c338] Remove abort handler once fired once

**Pull request merged**